File: WASMXamarin/WASMXamarin/WASMXamarin.iOS/InterfaceTestCell.cs
using AVFoundation;
using CoreAnimation;
using CoreMedia;
using Foundation;
using System;

using UIKit;

namespace WASMXamarin.iOS
{
    public partial class InterfaceTestCell : UITableViewCell
    {
        AVPlayer _player;
        AVPlayerLayer _playerLayer;
        AVAsset _asset;
        AVPlayerItem _playerItem;
        UIImageView image;

        public InterfaceTestCell(IntPtr handle) : base(handle)
        {
        }
        public InterfaceTestCell(String cellId) : base(UITableViewCellStyle.Default, cellId)
        {
        }

        public override void AwakeFromNib()
        {
            base.AwakeFromNib();
            _asset = AVAsset.FromUrl(NSUrl.FromFilename("video.mp4"));
            _playerItem = new AVPlayerItem(_asset);
            _player = new AVPlayer(_playerItem);

            image = new UIImageView(UIImage.FromFile("Icon-76.png"));
            label.Font = UIFont.FromName("Cochin-BoldItalic", 22f);
            label.TextColor = UIColor.FromRGB(127, 51, 0);
            label.BackgroundColor = UIColor.Clear;
            LayoutIfNeeded();
            _playerLayer = AVPlayerLayer.FromPlayer(_player);
            
            _player.ExternalPlaybackVideoGravity = AVLayerVideoGravity.ResizeAspectFill;
        }

        public override void LayoutSubviews()
        {
            base.LayoutSubviews();      

            image.Frame = new CoreGraphics.CGRect(Frame.Size.Width - 40, Frame.Size.Height - 40, 20, 20);
            image.ContentMode = UIViewContentMode.ScaleAspectFit;

            CABasicAnimation rotationAnimation = new CABasicAnimation();
            rotationAnimation.KeyPath = "transform.rotation.z";
            rotationAnimation.To = new NSNumber(Math.PI * 2);
            rotationAnimation.Duration = 1;
            rotationAnimation.Cumulative = true;
            rotationAnimation.RepeatCount = float.MaxValue;
            Add(image);
            image.Layer.AddAnimation(rotationAnimation, "rotationAnimation");            
        }

        public void UpdateCell(string labeltext)
        {
            label.Text = labeltext;
            _playerLayer.Frame = new CoreGraphics.CGRect(5, 30, Frame.Size.Width - 10, Frame.Size.Height - 40); ;
            playerView.Layer.AddSublayer(_playerLayer);
            _player.Play();
        }
    }
}

File: WASMiOS/WASMiOS/InterfaceTestCell.m
//
//  InterfaceTestCell.m
//  WASMiOS
//
//  Created by Pawel Grzmil on 17/01/16.
//  Copyright © 2016 Pawel Grzmil. All rights reserved.
//

#import "InterfaceTestCell.h"
@import AVKit;
@import AVFoundation;

@interface InterfaceTestCell()
@property (weak, nonatomic) IBOutlet UILabel *label;
@property (weak, nonatomic) IBOutlet UIView *playerView;
@property (weak, nonatomic) IBOutlet UIImageView *icon;

@end

@implementation InterfaceTestCell
AVAsset* avAsset;
AVPlayerItem* avPlayerItem;
AVPlayerLayer* avPlayerLayer;
AVPlayer* avPlayer;

-(void)awakeFromNib{
    [super awakeFromNib];
    
    NSString *path = [[NSBundle mainBundle] pathForResource:@"video" ofType:@"mp4"];
    NSURL *url = [NSURL fileURLWithPath:path];
    avAsset = [AVAsset assetWithURL:url];
    avPlayerItem =[[AVPlayerItem alloc]initWithAsset:avAsset];
    avPlayer = [[AVPlayer alloc]initWithPlayerItem:avPlayerItem];
    avPlayerLayer =[AVPlayerLayer playerLayerWithPlayer:avPlayer];
    avPlayer.externalPlaybackVideoGravity = AVLayerVideoGravityResizeAspectFill;

    CGRect a= CGRectMake(-105,0, self.frame.size.width-10, self.frame.size.height-40);
    [avPlayerLayer setFrame:a];
    [self.playerView.layer addSublayer:avPlayerLayer];
}

-(void)layoutSubviews{
    [super layoutSubviews];
    
    CABasicAnimation* rotationAnimation;
    rotationAnimation = [CABasicAnimation animationWithKeyPath:@"transform.rotation.z"];
    rotationAnimation.toValue = [NSNumber numberWithFloat: M_PI * 2.0 ];
    rotationAnimation.duration = 1;
    rotationAnimation.cumulative = YES;
    rotationAnimation.repeatCount = CGFLOAT_MAX;
    [self.icon.layer addAnimation:rotationAnimation forKey:@"rotationAnimation"];
}

-(void)updateCell:(NSString*)labelText{
    self.label.text = labelText;
    [avPlayer seekToTime:kCMTimeZero];
    [avPlayer play];
}

@end
